WordPress/Classes/ViewRelated/Domains/Register/RegisterDomainDetailsViewController.swift

```diff
@@ -0,0 +1,183 @@
+import UIKit
+
+class RegisterDomainDetailsViewController: UITableViewController {
+
+    private enum Constants {
+        static let estimatedRowHeight: CGFloat = 62
+    }
+
+    private enum Section: Int {
```

Comment:
Something about duplicating information between here and the `ImmuTable` seems off to me. I haven't tried if it actually works, but maybe subclassing ` ImmuTableSection` that would let you hold both `title` and `subtitle` would let you get rid of this enum here?

Reply:
I couldn't understand the relation between this enum and subclassing ImmuTableSection. This enum is used for constructing section headers and that's all(I am putting its usage below). Could you please give more details? Thanks

open override func tableView(_ tableView: UITableView, viewForHeaderInSection section: Int) -> UIView? {
        guard let sectionType = Section(rawValue: section) else {
            return nil
        }
        switch sectionType {
        case .privacyProtection:
            return privacyProtectionSectionHeader()
        case .contactInfo:
            return contactInformationSectionHeader()
        default:
            break
        }
        return nil
    }

Reply:
Hey, sorry if this was confusing. 

What I meant was: with your code as it is now, the information about sections is contained in two places: one of it is `RegisterDomainDetailsViewController.tableViewModel()` function, and the other is the `RegisterDomainDetailsViewController.Sections` enum. 
This is  somewhat redundant and future modifications of this table will require modifying code in two places.

As you say, the only place you're using the `RegisterDomainDetailsViewController.Sections` enum, is for constructing the header — which consists of `title` and `description`. `ImmuTableSection` already has a property called `headerText`, so we're just missing a place to put in description. That's where my idea for a subclass comes in!

If we had something like 
```swift
struct RegisterDomainSection: ImmuTableSection {
    let descriptionText: String?
}
```
then your section header code could just look like this (pseudo-code, haven't checked if it compiles):
```swift 
open override func tableView(_ tableView: UITableView, viewForHeaderInSection section: Int) -> UIView? {
    guard let section = tableViewModel()[section] as? RegisterDomainSection, let header = section.headerText, let description = section.descriptionText else {
        return nil
    }
    return sectionHeader(title: header, description: description)
}
```

This way, there's only ever one place where we're declaring what the table looks like — in the `tableViewModel()` method defining the `ImmuTable` model :)


Let me know if this clears things up for you :)

Reply:
I think I got it now, I'll work on that

Reply:
I've checked this but since ImmuTableSection is a struct we can not inherit it